packages/@aws-cdk/aws-nodejs-aspect/lib/nodejs-aspect.ts

```diff
@@ -0,0 +1,137 @@
+    targetNode.runtime = this.targetRuntime;
+  }
+
+  private handleCustomResourceProvider(provider: cdk.CustomResourceProviderBase): void {
+    const node = provider as cdk.CustomResourceProviderBase;
+    // only replace for nodejs case
+    if (this.isValidRuntime(node.runtime)) {
+      const targetnode = this.getChildFunctionNode(node);
+      targetnode.addPropertyOverride('Runtime', this.targetRuntime);
+    }
+  }
+
+  private handleProvider(provider: Provider): void {
+    const functionNodes = provider.node.children.filter((child) => child instanceof lambda.Function);
+    for ( var node of functionNodes) {
+      const targetNode = this.getChildFunctionNode(node);
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        targetNode.addPropertyOverride('Runtime', this.targetRuntime);
+      }
+
+      // onEvent Handlers
+      const onEventHandler = provider.onEventHandler as lambda.Function;
+      const onEventHandlerRuntime = this.getChildFunctionNode(onEventHandler);
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        onEventHandlerRuntime.addPropertyOverride('Runtime', this.targetRuntime);
+      }
+
+      //isComplete Handlers
+      // Handlers
+      const isCompleteHandler = provider.isCompleteHandler as lambda.Function;
+      const isCompleteHandlerRuntime = this.getChildFunctionNode(isCompleteHandler);
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        isCompleteHandlerRuntime.addPropertyOverride('Runtime', this.targetRuntime);
+      }
+    }
+  }
+
+  private handleReceiptRuleSet(ruleSet: ReceiptRuleSet): void {
+    const functionnode = ruleSet.node.findChild('DropSpam');
+    const ses = functionnode.node.findChild('Function') as lambda.CfnFunction;
+    if (ses.runtime && this.isValidRuntime(ses.runtime)) {
+      ses.addPropertyOverride('Runtime', this.targetRuntime);
+    }
+  }
+
+  /**
+ *Runtime Validation
+ *
+ */
+  private isValidRuntime(runtime: string) {
+    const family = this.getRuntimeFamily(runtime);
+    return family === lambda.RuntimeFamily.NODEJS;
+  }
+
+  private getRuntimeProperty(node: cdk.CfnResource) {
+    return (node.getResourceProperty('runtime') || node.getResourceProperty('Runtime')) as string;
+  }
+
+  private getChildFunctionNode(node: Construct) : cdk.CfnResource {
+    const childNode = node.node.children.find((child) => cdk.CfnResource.isCfnResource(child) && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+    return childNode;
+  }
+
+  private getRuntimeFamily(runtime: string) {
+    switch (runtime) {
+      case 'nodejs18.x':
+        return lambda.Runtime.NODEJS_18_X.family;
+      default:
+        return 'Unsupported';
+    }
+  }
+
+}
+
+/**
+ * RuntimeAspect
+ */
+export class RuntimeAspect extends RuntimeAspectsBase {
+  constructor(key: string) {
+    super(key);
+  }
+}
+
+export class NodeJsAspect {
+  public static modifyRuntimeTo(key: lambda.Runtime) {
```

Comment:
See my above comment. I think a better user experience here is to do all the construction for them. We can expose options like `nodeJs20` and then we can to all the construction like passing the associated `Runtime` for the user. This also prevents the user from breaking themselves, like passing node 16 which would be breaking since it only comes with SDK v2.

Reply:
yes, was trying with implementation earlier too but it was not reflecting the updated nodejs version in template this way, gave it a second attempt and it seems to be working now updated in latest commit.